eng/sendtohelix.proj

```diff
@@ -68,11 +68,15 @@
 
 
   <PropertyGroup Condition="'$(HelixCommand)' == ''">
-    <!-- For windows we need to use call, since the command is going to be called from a bat script created by Helix
-    and we exit /b at the end of RunTests.cmd, Helix runs some other commands after ours within the bat script,
-    if we don't use call, then we cause the parent script to exit, and anything after will not be executed. -->
-    <HelixCommand Condition="'$(TargetsWindows)' == 'true'">call RunTests.cmd %HELIX_CORRELATION_PAYLOAD%</HelixCommand>
-    <HelixCommand Condition="'$(TargetsWindows)' != 'true'">./RunTests.sh $HELIX_CORRELATION_PAYLOAD</HelixCommand>
+    <!--
+      For windows we need to use call, since the command is going to be called from a bat script created by Helix
+      and we exit /b at the end of RunTests.cmd, Helix runs some other commands after ours within the bat script,
+      if we don't use call, then we cause the parent script to exit, and anything after will not be executed.
+      The arguments passed in to the run script in order are the runtime directory, the dotnet root directory (for
+      helix submissions same as the runtime directory) and the global tools directory.
+    -->
+    <HelixCommand Condition="'$(TargetsWindows)' == 'true'">call RunTests.cmd %HELIX_CORRELATION_PAYLOAD% %HELIX_CORRELATION_PAYLOAD% %HELIX_CORRELATION_PAYLOAD%\tools</HelixCommand>
```

Comment:
why is ` %HELIX_CORRELATION_PAYLOAD% %HELIX_CORRELATION_PAYLOAD%` being passed twice?

Reply:
See the comment above.

> The arguments passed in to the run script in order are the runtime directory, the dotnet root directory (for helix submissions same as the runtime directory) and the global tools directory.

Reply:
Adding my edited comment above:

I see in the comment above: `runtime directory, the dotnet root directory`. I don't understand the difference.

Reply:
The runtime directory is the host and runtime that we use for testing (with the assets that we build live). The dotnet root is the host and runtime that we use for additional tooling like global tools. It's the one that is picked up by shims, again i.e. global tools.

Reply:
I guess the reason why it is duplicated is because for Helix it is the same guy, but on a local environment, someone could need/want a different DOTNET_ROOT, and use a local dotnet SDK where they have their tools installed?

Reply:
copying my response from the offline discussion with Eric:

dotnet folder = like the official dotnet folder, runtime + host + fxr
 
The first one is the dotnet folder that we use for testing. It contains our self built shared framework assemblies
The second one is the dotnet folder that we use for invoking tools like global or repo tools. We don't want to use our self built shared framework here as that could caues issues.
The third one is the tools folder where the global tools shims live in i.e. coverlet.exe and reportgenerator.exe
 
Now we have two flavours:
without helix: 1) corefx\.dotnet, 2) corefx\artifacts\bin\testhost\netcoreapp-...\, 3) corefx\artifacts\tools
with helix: 1) helixpayload-runtime, 2) helixpayload-runtime, 3) helixpayload-runtime\tools
 
for helix we use the same host but put both shared frameworks in that folder. the 9.9.9 which is for testing and the stable one, i.e. 2.1.3

Reply:
Makes sense. Thanks for the explaining in such detail.